src/MongoDB.Bson/Serialization/Conventions/ImmutableTypeClassMapConvention.cs

```diff
@@ -105,5 +106,48 @@ private bool CanWrite(PropertyInfo propertyInfo)
             // CanWrite gets true even if a property has only a private setter
             return propertyInfo.CanWrite && (propertyInfo.SetMethod?.IsPublic ?? false);
         }
+
+        private bool PropertyMatchesSomeCreatorParameter(BsonClassMap classMap, PropertyInfo propertyInfo)
+        {
+            foreach (var creatorMap in classMap.CreatorMaps)
+            {
+                if (creatorMap.MemberInfo is ConstructorInfo constructorInfo)
+                {
+                    if (PropertyMatchesSomeConstructorParameter(constructorInfo))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            // also map properties that match some constructor parameter that might be called by a derived class
+            foreach (var constructorInfo in classMap.ClassType.GetTypeInfo().GetConstructors(_constructorBindingFlags))
```

Comment:
I find moving the binding flag constants to a variable unhelpful. 

It means we can't fully understand what line 124 does without tracking down what the *current* value of the variable is.

Reply:
I would say it's helpful, it means that all searching for constructors in this file work in the same way regardless of what lines in this file we're looking in